persistent/Database/Persist/Quasi.hs

```diff
@@ -109,21 +355,26 @@ lowerCaseSettings = defaultPersistSettings
 
 -- | Parses a quasi-quoted syntax into a list of entity definitions.
 parse :: PersistSettings -> Text -> [EntityDef]
-parse ps = parseLines ps
-      . removeSpaces
-      . filter (not . empty)
-      . map tokenize
-      . T.lines
+parse ps = parseLines ps . preparse
+
+preparse :: Text -> [Line]
+preparse =
+    removeSpaces
+        . filter (not . empty)
+        . map tokenize
+        . T.lines
```

Comment:
i experienced confusing behavior so i factored a lot of functions out and wrote many tests